osu.Game.Rulesets.Catch/UI/DefaultHitExplosion.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Utils;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Catch.Skinning.Default;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.UI
+{
+    public class DefaultHitExplosion : CatchHitExplosion
+    {
+        private readonly FadePiece largeFaint;
+        private readonly FadePiece smallFaint;
+        private readonly GlowPiece directionalGlow1;
+        private readonly GlowPiece directionalGlow2;
+
+        private Color4 lastColor;
+
+        public DefaultHitExplosion()
+        {
+            Size = new Vector2(20);
+            Anchor = Anchor.TopCentre;
+            Origin = Anchor.BottomCentre;
+
+            const float initial_height = 10;
+
+            InternalChildren = new Drawable[]
+            {
+                directionalGlow1 = new GlowPiece
+                {
+                    Size = new Vector2(0.01f, initial_height)
+                },
+                directionalGlow2 = new GlowPiece
+                {
+                    Size = new Vector2(0.01f, initial_height),
+                },
+                largeFaint = new FadePiece(),
+                smallFaint = new FadePiece()
+            };
+        }
+
+        private void onColourChanged()
+        {
+            const float roundness = 100;
+
+            largeFaint.EdgeEffect = new EdgeEffectParameters
+            {
+                Type = EdgeEffectType.Glow,
+                Colour = Interpolation.ValueAt(0.1f, ObjectColour, Color4.White, 0, 1).Opacity(0.3f),
+                Roundness = 160,
+                Radius = 200,
+            };
+
+            smallFaint.EdgeEffect = new EdgeEffectParameters
+            {
+                Type = EdgeEffectType.Glow,
+                Colour = Interpolation.ValueAt(0.6f, ObjectColour, Color4.White, 0, 1),
+                Roundness = 20,
+                Radius = 50,
+            };
+
+            directionalGlow1.EdgeEffect = directionalGlow2.EdgeEffect = new EdgeEffectParameters
+            {
+                Type = EdgeEffectType.Glow,
+                Colour = Interpolation.ValueAt(0.4f, ObjectColour, Color4.White, 0, 1),
+                Roundness = roundness,
+                Radius = 40,
+            };
+        }
+
+        public override void Animate()
+        {
+            Scale = new Vector2(HitObject.Scale);
+
+            const double duration = 400;
+
+            if (!(HitObject is Droplet))
```

Comment:
This is new (i.e. it wasn't there in the original implementation). Not sure about this change, do players prefer no explosions on droplets? Generally this came out of left field a bit, I don't recall turning explosions off on droplets on the default skin being discussed at any point.